go.mod

```diff
@@ -0,0 +1,114 @@
+	github.com/lafriks/xormstore v1.0.0
+	github.com/lib/pq v1.0.0
+	github.com/lunny/dingtalk_webhook v0.0.0-20171025031554-e3534c89ef96
+	github.com/markbates/goth v1.49.0
+	github.com/mattn/go-sqlite3 v1.10.0
+	github.com/matttproud/golang_protobuf_extensions v1.0.1 // indirect
+	github.com/mcuadros/go-version v0.0.0-20171003094716-88e56e02bea1
+	github.com/microcosm-cc/bluemonday v0.0.0-20161012083705-f77f16ffc87a
+	github.com/mschoch/smat v0.0.0-20160514031455-90eadee771ae // indirect
+	github.com/msteinert/pam v0.0.0-20151204160544-02ccfbfaf0cc
+	github.com/nfnt/resize v0.0.0-20160724205520-891127d8d1b5
+	github.com/philhofer/fwd v1.0.0 // indirect
+	github.com/pquerna/otp v0.0.0-20160912161815-54653902c20e
+	github.com/prometheus/client_golang v0.9.0
+	github.com/prometheus/client_model v0.0.0-20180712105110-5c3871d89910 // indirect
+	github.com/prometheus/common v0.0.0-20181020173914-7e9e6cabbd39 // indirect
+	github.com/prometheus/procfs v0.0.0-20181005140218-185b4288413d // indirect
+	github.com/russross/blackfriday v0.0.0-20180428102519-11635eb403ff
+	github.com/satori/go.uuid v1.2.0
+	github.com/sergi/go-diff v1.0.0
+	github.com/shurcooL/sanitized_anchor_name v0.0.0-20160918041101-1dba4b3954bc // indirect
+	github.com/steveyen/gtreap v0.0.0-20150807155958-0abe01ef9be2 // indirect
+	github.com/stretchr/testify v1.2.2
+	github.com/tinylib/msgp v0.0.0-20180516164116-c8cf64dff200 // indirect
+	github.com/tstranex/u2f v1.0.0
+	github.com/urfave/cli v0.0.0-20161102131801-d86a009f5e13
+	github.com/willf/bitset v0.0.0-20180426185212-8ce1146b8621 // indirect
+	github.com/yohcop/openid-go v0.0.0-20160914080427-2c050d2dae53
+	golang.org/x/crypto v0.0.0-20180904163835-0709b304e793
+	golang.org/x/net v0.0.0-20181023162649-9b4f9f5ad519
+	golang.org/x/oauth2 v0.0.0-20181101160152-c453e0c75759 // indirect
+	golang.org/x/sync v0.0.0-20180314180146-1d60e4601c6f
+	golang.org/x/sys v0.0.0-20181026144532-2772b66316d2
+	golang.org/x/text v0.3.0
+	gopkg.in/alexcesaro/quotedprintable.v3 v3.0.0-20150716171945-2caba252f4dc // indirect
+	gopkg.in/asn1-ber.v1 v1.0.0-20150924051756-4e86f4367175 // indirect
+	gopkg.in/bufio.v1 v1.0.0-20140618132640-567b2bfa514e // indirect
+	gopkg.in/editorconfig/editorconfig-core-go.v1 v1.2.0
+	gopkg.in/gomail.v2 v2.0.0-20160411212932-81ebce5c23df
+	gopkg.in/ini.v1 v1.31.1
+	gopkg.in/ldap.v2 v2.4.1
+	gopkg.in/macaron.v1 v1.2.4
+	gopkg.in/redis.v2 v2.3.2 // indirect
+	gopkg.in/src-d/go-billy.v4 v4.3.0 // indirect
+	gopkg.in/src-d/go-git.v4 v4.8.0 // indirect
+	gopkg.in/testfixtures.v2 v2.5.0
+	gopkg.in/yaml.v2 v2.0.0-20160928153709-a5b47d31c556 // indirect
+	strk.kbt.io/projects/go/libravatar v0.0.0-20160628055650-5eed7bff870a
+)
+
+replace (
+	github.com/boltdb/bolt v1.3.1 => github.com/go-gitea/bolt v1.3.1-0.20170420010917-ccd680d8c1a0
+	github.com/denisenkom/go-mssqldb v0.0.0-20181014144952-4e0d7dc8888f => github.com/denisenkom/go-mssqldb v0.0.0-20161128230840-e32ca5036449
```

Comment:
Why did you specify a replace by an old version ? Or is it generated by the go mod command ?

Reply:
Want to know too, we can also upgrade this driver on this PR if necessary.

Reply:
@sapk Done.

Reply:
@sapk @lunny By the way, that would probably not work because the Go module seems to always infer to the 20181014 revision.

Reply:
I use `go got -u` as you suggested above, and tried manually edited go.mod. But it always turns into the 20181014 version after `make vendor`.

Reply:
@sapk @lunny Looks like this won't work. I'll revert the change.

The `require` directive only specifies the minimum version. And as far as I know,  `mssqldb` does not follow the semvar convention, that could be the cause of the problem (or a bug in Go module, idk).

Reply:
@typeless that must be xorm that use the version 20181014 https://github.com/go-xorm/xorm/blob/master/go.mod#L6 so you are right with the replace that will catch this dep.